File: backend/app/schemas/progress.py
from pydantic import BaseModel


class MilestoneItem(BaseModel):
    key: str
    label: str
    done: bool


class MilestonesResponse(BaseModel):
    total: int
    completed: int
    items: list[MilestoneItem]


class AchievementItem(BaseModel):
    key: str
    label: str


class AchievementsResponse(BaseModel):
    total: int
    unlocked: int
    latest: AchievementItem | None


class ProgressResponse(BaseModel):
    strategies_count: int
    strategy_versions_count: int
    completed_backtests_count: int
    milestones: MilestonesResponse
    achievements: AchievementsResponse
    next_steps: list[str]

File: frontend/src/types/progress.ts
export interface MilestoneItem {
  key: string;
  label: string;
  done: boolean;
}

export interface MilestonesResponse {
  total: number;
  completed: number;
  items: MilestoneItem[];
}

export interface AchievementItem {
  key: string;
  label: string;
}

export interface AchievementsResponse {
  total: number;
  unlocked: number;
  latest: AchievementItem | null;
}

export interface ProgressResponse {
  strategies_count: number;
  strategy_versions_count: number;
  completed_backtests_count: number;
  milestones: MilestonesResponse;
  achievements: AchievementsResponse;
  next_steps: string[];
}
